src/lib.rs

```diff
@@ -0,0 +1,1166 @@
+    ///     v1.and(&v2);
+    ///     assert_eq!(v1, vob![true, false, false]);
+    /// }
+    /// ```
+    pub fn and(&mut self, other: &Vob<T>) -> bool {
+        if self.bits != other.bits {
+            panic!("Cannot 'and' two Vobs of different length ({}  {})", self.bits, other.bits);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov & *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// For each bit in this Vob, `or` it with the corresponding bit in `other`, returning `true`
+    /// if this led to any changes or `false` otherwise. The two Vobs must have the same number of
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v1 = vob![true, false, false];
+    ///     let v2 = vob![false, true, false];
+    ///     v1.or(&v2);
+    ///     assert_eq!(v1, vob![true, true, false]);
+    /// }
+    /// ```
+    pub fn or(&mut self, other: &Vob<T>) -> bool {
+        if self.bits != other.bits {
+            panic!("Cannot 'or' two Vobs of different length ({}  {})", self.bits, other.bits);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov | *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// For each bit in this Vob, `xor` it with the corresponding bit in `other`, returning `true`
+    /// if this led to any changes or `false` otherwise. The two Vobs must have the same number of
+    /// bits.
+    ///
+    /// # Panics
+    ///
+    /// If the two Vobs are of different length.
+    ///
+    /// # Examples
+    /// ```
+    /// #[macro_use] extern crate vob;
+    /// fn main() {
+    ///     let mut v1 = vob![true, false, true];
+    ///     let v2 = vob![false, true, true];
+    ///     v1.xor(&v2);
+    ///     assert_eq!(v1, vob![true, true, false]);
+    /// }
+    /// ```
+    pub fn xor(&mut self, other: &Vob<T>) -> bool {
+        if self.bits != other.bits {
+            panic!("Cannot 'xor' two Vobs of different length ({}  {})", self.bits, other.bits);
+        }
+        let mut chngd = false;
+        for (sb, ob) in self.vec.iter_mut().zip(other.vec.iter()) {
+            let ov = *sb;
+            let nv = ov ^ *ob;
+            if ov != nv {
+                chngd = true;
+            }
+            *sb = nv;
+        }
+        self.mask_last_block();
+        chngd
+    }
+
+    /// We guarantee that the last storage block has no bits set past the "last" bit: this function
```

Comment:
Why is this useful? This guarantee an optimisation?

Reply:
We allow users to iterate over the blocks: we don't want to see them bits set beyond `self.len()`, otherwise they'll get confused.

Reply:
I see. Next silly question, why expose block to the user?

Reply:
Operating on blocks is *massively* faster than on individual bits. `Vob` provides `and`, `or`, and `xor` as common cases, but there are others. `cfgrammar` implements its own `intersect` which iterates over blocks, for example. [Should I include `intersect` in `Vob`? I'm not sure. It's difficult to know where to stop...]

Reply:
I guess I would expect an interface which exposes a slice which cannot extend beyond the length of the vob. Might be hard to implement?

Reply:
It's also convenient to maintain this guarantee internally, so it wouldn't really make sense to only do it in an interator.